src/encodings/rle.rs

```diff
@@ -189,6 +189,14 @@ impl RleEncoder {
     self.bit_writer.consume()
   }
 
+  /// Borrow equivalent of the `consume` method.
+  /// Call `clear()` after invoking this method.
+  #[inline]
+  pub fn flush_buffer(&mut self) -> &[u8] {
+    self.bit_writer.flush();
```

Comment:
can we also add a `flush_buffer()` for `bit_writer`, which flushes the buffer and return its borrowed reference? this pattern occurs a few times in `encoding.rs`.

Also, maybe this `flush_buffer` should call `flush` first? it's a little confusing that we need to call `flush` and then call `flush_buffer`.

Reply:
Yes, I will add the method and update occurrences.